ipa-core/src/net/test.rs

```diff
@@ -91,30 +185,40 @@ pub fn server_config_https(
     }
 }
 
-#[derive(Default)]
 pub struct TestConfigBuilder {
-    ports: Option<[u16; 3]>,
+    /// Can be empty meaning that free ports should be obtained from the operating system.
+    /// One ring per shard in a helper (sharding factor). For each ring we need 3 shard and
+    /// 3 mpc ports.
+    ports_by_ring: Vec<Ports<Vec<u16>>>,
+    /// Describes the number of shards per helper
+    sharding_factor: usize,
```

Comment:
That was intended because you're getting 3x hosts for the factor.

If you still think shard_count is easier I can do that, but this factor affects all 3 helpers.

Reply:
then `/// Describes the number of shards per helper`  confuses me even more

Reply:
Why? it says per helper aka 3x